recipes/box2d/all/conanfile.py

```diff
@@ -78,6 +78,7 @@ def source(self):
     def generate(self):
         tc = CMakeToolchain(self)
         tc.cache_variables["CMAKE_POLICY_DEFAULT_CMP0077"] = "NEW"
+        tc.variables["CMAKE_CXX_FLAGS"] = "-O3 -ftree-vectorize -fopt-info-vec -Wno-pass-failed"
```

Comment:
Hi @skhaz - thanks for your contribution.

Doing it this way does not take into account the compiler - they are passed unconditionally to all compilers regardless of whether they understand those particular flags. Additionally, `-O3` would trample and override the user's build_type setting - since it is being applied unconditionally.

This looks like a bug in the upstream recipe itself - I would first ensure that it is, at the very least, reported upstream to the maintainers: https://github.com/erincatto/box2d

In this instance, however I would possibly advise simply setting 
```
tc.cache_variables["CMAKE_COMPILE_WARNING_AS_ERROR"] = False
```

since this is failing on emscriptem because this setting is too strict

Reply:
Thank you, I have updated my pull request and raised an issue on box2d repository.